### PR TITLE
Provide error message when allow_stdio_access creates an undecideable error

### PR DESCRIPTION
**Pull request by SeanTAllen**
If two containers are indistinguishable at create_container or exec_external time except for their `allow_stdio_access` value, then we are unable to proceed as we don't know if access should be allowed or not.

Prior to this commit, no error message at all was displayed when this was hit making it almost impossible for someone not "in the know" to diagnose.

This first version gives a very simple error message that will be improved with more information in later commits.

Signed-off-by: Sean T. Allen <seanallen@microsoft.com>

### CODE REVIEW

File: pkg/securitypolicy/regopolicy_test.go [hunk 1]
```diff
@@ -3957,6 +3957,79 @@ func Test_Rego_EnvListGetsRedacted(t *testing.T) {
 	}
 }
 
+func Test_Rego_Enforce_CreateContainer_ConflictingAllowStdioAccessHasErrorMessage(t *testing.T) {
+	constraints := generateConstraints(testRand, 1)
+	constraints.containers[0].AllowStdioAccess = true
+
+	// create a "duplicate" as far as create container is concerned excep for
+	// a different "AllowStdioAccess" value
+	duplicate := &securityPolicyContainer{
```
Comment (matajoh) [inline pkg/securitypolicy/regopolicy_test.go:3966]: FWIW I added a clone method for this purpose.

File: pkg/securitypolicy/regopolicy_test.go [hunk 2]
```diff
@@ -3957,6 +3957,79 @@ func Test_Rego_EnvListGetsRedacted(t *testing.T) {
...
+	if !strings.Contains(err.Error(), "containers only distinguishable by allow_stdio_access") {
+		t.Fatal("No error message given for conflicting allow_stdio_access on otherwise 'same' containers")
+	}
+}
+
+func Test_Rego_ExecExternalProcessPolicy_ConflictingAllowStdioAccessHasErrorMessage(t *testing.T) {
+	constraints := generateConstraints(testRand, 1)
+	process := generateExternalProcess(testRand)
+	process.allowStdioAccess = false
+	duplicate := &externalProcess{
```
Comment (matajoh) [inline pkg/securitypolicy/regopolicy_test.go:4001]: Again, there's a clone method that does this.

File: pkg/securitypolicy/securitypolicyenforcer_rego.go [hunk 3]
```diff
@@ -318,6 +318,11 @@ func (policy *regoEnforcer) getReasonNotAllowed(enforcementPoint string, input i
 
 func (policy *regoEnforcer) redactSensitiveData(input inputData) inputData {
 	if v, k := input["envList"]; k {
+		newInput := make(inputData)
```
Comment (matajoh) [inline pkg/securitypolicy/securitypolicyenforcer_rego.go:321]: This doesn't seem germane to this PR?

Comment (SeanTAllen) [reply pkg/securitypolicy/securitypolicyenforcer_rego.go:321]: Tests can't pass without this  fix.

File: pkg/securitypolicy/regopolicy_test.go (hunk 2 above)
Comment (SeanTAllen) [reply pkg/securitypolicy/regopolicy_test.go:4001]: ah yes I should use clone here.

File: pkg/securitypolicy/regopolicy_test.go (hunk 1 above)
Comment (SeanTAllen) [reply pkg/securitypolicy/regopolicy_test.go:3966]: This isnt quite a clone. It was an intentional not clone (unlike the exec external one). As it isn't copying everything over. This felt more obvious. Do you think I should clone and then replace the "not germane signals and exec processes?

File: pkg/securitypolicy/securitypolicyenforcer_rego.go (hunk 3 above)
Comment (SeanTAllen) [reply pkg/securitypolicy/securitypolicyenforcer_rego.go:321]: we'll pick this up with a rebase im about to do.

Comment (SeanTAllen) [reply pkg/securitypolicy/securitypolicyenforcer_rego.go:321]: this is how i found the bug in redact sensitive data.

File: pkg/securitypolicy/regopolicy_test.go (hunk 1 above)
Comment (matajoh) [reply pkg/securitypolicy/regopolicy_test.go:3966]: I misread the code here, didn't fully see that you were intentionally keeping somethings different. It should stay as is then.

File: pkg/securitypolicy/securitypolicyenforcer_rego.go (hunk 3 above)
Comment (matajoh) [reply pkg/securitypolicy/securitypolicyenforcer_rego.go]: Ah gotcha!

**Pull request closed without merging**